lib/Monad_Commute.thy

```diff
@@ -0,0 +1,195 @@
+(*
+ * Copyright 2023, Proofcraft Pty Ltd
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ *)
+
+(* The monad_commute predicate + lemmas. *)
```

Comment:
no idea what monad_commute is for ... again if we factor it out, should we say anything?

Reply:
You shouldn't punish people doing cleanup by demanding more cleanup :-)

I didn't write this predicate and I'm not sure where it's used, but I thought the name and definition are obvious enough: it says when two monads commute. Without tracking down where it's used, I'm not sure what else I would say about it.

It does look like yet another instance of the general oblivious /isoatedThreadAction etc idea, so eventually we should replace all these with just one concept, but this PR is just about dependencies and factoring things out.